src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -63,13 +64,13 @@ internal sealed class RestoreOperationLogger : ILogger
 
         public async Task StartAsync(
             RestoreOperationSource operationSource,
-            ErrorListProvider errorListProvider,
+            Lazy<ErrorListTableDataSource> errorListDataSource,
```

Comment:
This is not on any of those critical paths. Logger is instantiated every time it's requested, IE on every actual restore being started. Since you made the error list a MEF component it'd be right thing to move this to importing constructor.